src/model/app.rs

```diff
@@ -36,23 +51,117 @@ impl App {
         self.app_mode
     }
 
+    pub fn set_app_mode(&mut self, app_mode: AppMode) {
+        self.app_mode = app_mode;
+    }
+
     pub fn get_text_as_iter(&self) -> Vec<String> {
         vec![self.buffer.as_str()]
     }
 
+    // Originally implemented with get_text_as_iter using a match depending on the app_mode
+    // but decided to separate it out and write another method because accessing a buffer
+    // should be independent of app_mode.
+    // We can discuss the naming of this method if required
+    pub fn get_command_buffer_text_as_iter(&self) -> Vec<String> {
+        vec![self.command_buffer.as_str()]
+    }
+
+    // This method was written to make the View as "dumb" as possible
+    // The App will handle which text is to be shown
+    pub fn get_text_based_on_mode(&self) -> Vec<String> {
+        match self.app_mode() {
+            AppMode::Edit => self.get_text_as_iter(),
+            AppMode::Command(_) => self.get_command_buffer_text_as_iter(),
+        }
+    }
+
     pub fn add_char(&mut self, c: char) {
-        self.buffer.insert(&mut self.cursor_main, c);
+        match self.app_mode() {
+            AppMode::Edit => self.buffer.insert(&mut self.cursor_main, c),
+            AppMode::Command(_) => self.command_buffer.insert(&mut self.cursor_main, c),
+        }
     }
 
     pub fn remove_char(&mut self) {
-        self.buffer.remove(&mut self.cursor_main);
+        match self.app_mode() {
+            AppMode::Edit => self.buffer.remove(&mut self.cursor_main),
+            AppMode::Command(_) => self.command_buffer.remove(&mut self.cursor_main),
+        }
     }
 
     pub fn move_cursor_left(&mut self) {
-        self.buffer.move_cursor_left(&mut self.cursor_main);
+        match self.app_mode() {
+            AppMode::Edit => self.buffer.move_cursor_left(&mut self.cursor_main),
+            AppMode::Command(_) => self.command_buffer.move_cursor_left(&mut self.cursor_main),
+        }
     }
 
     pub fn move_cursor_right(&mut self) {
-        self.buffer.move_cursor_right(&mut self.cursor_main);
+        match self.app_mode() {
+            AppMode::Edit => self.buffer.move_cursor_right(&mut self.cursor_main),
+            AppMode::Command(_) => self.command_buffer.move_cursor_right(&mut self.cursor_main),
+        }
+    }
+
+    pub fn save_file(&mut self) {
+        assert!(self.app_mode() == AppMode::Command(CommandMode::Write));
+
+        let file_path = self.get_command_buffer_text_as_iter().join("");
+
+        // Get from normal buffer
+        let text_to_save = self.get_text_as_iter().join("");
+
+        fs::write(file_path, text_to_save).expect("Unable to write file");
+
+        // The current implementation will quit after saving. This is here
+        // for the time when we are going to implement saving without quitting
+        self.set_app_mode(AppMode::Edit);
+    }
+
+    pub fn open_file(&mut self) {
+        assert!(self.app_mode() == AppMode::Command(CommandMode::Read));
```

Comment:
Also change this to `assert_eq!`. I kinda forgot this macro existed bcs I was writing outside the test methods ahahah ._.